### PR TITLE
Checklock: Customized Kernel, KernelApp, KernelSpecManager

### PR DESCRIPTION
**Pull request by KathiaBarahona**
Documentation will be pending.

For now, this might help for visualization purpose:

image

### CODE REVIEW

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/settings.py [hunk 1]
```diff
@@ -28,20 +28,20 @@ class Config:
...
 
         if self.bq_db:
-            mapping[SupportedEngine.BIGQUERY] = self.bq_db
+            mapping[SupportedDBEngine.BIGQUERY] = self.bq_db
 
         if self.athena_db:
-            mapping[SupportedEngine.ATHENA] = self.athena_db
+            mapping[SupportedDBEngine.ATHENA] = self.athena_db
 
         return mapping
```
Comment (jansentom) [inline modelhub/tests_modelhub/checklock_holmes/checklock_holmes/settings.py:42]: `if len(mapping) == 0: raise ...` ?

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/kernel.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    aliases = Dict(
+        {
+            **kernel_aliases,
+            'notebook': 'IPKernelApp.notebook_name',
+            'db_engine': 'IPKernelApp.db_engine',
+            'check_id': 'IPKernelApp.check_id',
+        }
+    )
+
+    notebook_name: Unicode = Unicode(help='The name of the notebook been executed.').tag(config=True)
```
Comment (jansentom) [inline modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/kernel.py:37]:
```suggestion
    notebook_name: Unicode = Unicode(help='The name of the notebook being executed.').tag(config=True)
```

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/manager.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        self._kernel_spec = self._get_kernel_spec(
+            notebook_name, db_engine, check_id,
+        )
+
+    @staticmethod
+    def _get_kernel_spec(
+        notebook_name: str, db_engine: SupportedDBEngine, check_id: UUID
+    ) -> KernelSpec:
+        argv = [
+            "python", "-m", "checklock_holmes.checklock_kernel.kernel", "-f", "{connection_file}",
```
Comment (jansentom) [inline modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/manager.py:35]: on my system, this should be
```suggestion
            "python3", "-m", "checklock_holmes.checklock_kernel.kernel", "-f", "{connection_file}",
```

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_nb_client/notebook_client.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    ):
+        km = AsyncChecklockKernelManager(notebook_name, db_engine, check_id, **kwargs)
+        super(ChecklockNBClient, self).__init__(nb_man, km=km, **kwargs)
+
+    async def async_execute(self, reset_kc: bool = False, **kwargs: Any) -> nbformat.NotebookNode:
+        nb = await super().async_execute(reset_kc, **kwargs)
+
+        # shutdown the Kernel after finishing executing the notebook,
+        # this way we avoid having dangling processes
+        await self._async_cleanup_kernel()
```
Comment (jansentom) [inline modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_nb_client/notebook_client.py:44]: should this be wrapped in `try .. finally`?

**Review by jansentom — COMMENTED** [review]
This was a bit of a "if it works it works" review. Left some comments though.

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/settings.py (hunk 1 above)
Comment (KathiaBarahona) [reply modelhub/tests_modelhub/checklock_holmes/checklock_holmes/settings.py:42]: actually, yes

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_nb_client/notebook_client.py (hunk 4 above)
Comment (KathiaBarahona) [reply modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_nb_client/notebook_client.py:44]: good catch :+1:

Comment (KathiaBarahona) [reply modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_nb_client/notebook_client.py]: https://github.com/objectiv/objectiv-analytics/pull/1406/commits/21f56981c5f0fde83a7b9e36f9d091a46f6676ce

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/manager.py (hunk 3 above)
Comment (KathiaBarahona) [reply modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/manager.py]: https://github.com/objectiv/objectiv-analytics/pull/1406/commits/21f56981c5f0fde83a7b9e36f9d091a46f6676ce

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/kernel.py (hunk 2 above)
Comment (KathiaBarahona) [reply modelhub/tests_modelhub/checklock_holmes/checklock_holmes/checklock_kernel/kernel.py]: https://github.com/objectiv/objectiv-analytics/pull/1406/commits/21f56981c5f0fde83a7b9e36f9d091a46f6676ce

File: modelhub/tests_modelhub/checklock_holmes/checklock_holmes/settings.py (hunk 1 above)
Comment (KathiaBarahona) [reply modelhub/tests_modelhub/checklock_holmes/checklock_holmes/settings.py:42]: tho it is handled here:
https://github.com/objectiv/objectiv-analytics/blob/1a91ba9c18914e9c02cccbe5c674c0cd29003044/modelhub/tests_modelhub/checklock_holmes/checklock-holmes.py#L83-L88

**Pull request merged**